bundles/org.openhab.binding.coolmasternet/src/main/java/org/openhab/binding/coolmasternet/internal/handler/HVACHandler.java

```diff
@@ -49,107 +49,130 @@
-                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
-                        String.format("Could not connect to CoolMasterNet unit %s:%d", config.host, config.port));
-            } else {
-                if (channel.endsWith(ON) && command instanceof OnOffType) {
-                    OnOffType onoff = (OnOffType) command;
-                    controller.sendCommand(String.format("%s %s", onoff == OnOffType.ON ? "on" : "off", uid));
-                } else if (channel.endsWith(SET_TEMP) && command instanceof DecimalType) {
-                    DecimalType temp = (DecimalType) command;
-                    controller.sendCommand(String.format("temp %s %s", uid, temp));
-                } else if (channel.endsWith(MODE) && command instanceof StringType) {
-                    /* the mode value in the command is the actual CoolMasterNet protocol command */
-                    controller.sendCommand(String.format("%s %s", command, uid));
-                } else if (channel.endsWith(FAN_SPEED) && command instanceof StringType) {
-                    controller.sendCommand(String.format("fspeed %s %s", uid, command));
-                } else if (channel.endsWith(LOUVRE) && command instanceof StringType) {
-                    controller.sendCommand(String.format("swing %s %s", uid, command));
-                } else if (command instanceof RefreshType) {
-                    refresh();
-                }
+            if (channel.endsWith(ON) && command instanceof OnOffType) {
+                final OnOffType onoff = (OnOffType) command;
+                controller.sendCommand(String.format("%s %s", onoff == OnOffType.ON ? "on" : "off", uid));
+            } else if (channel.endsWith(SET_TEMP) && command instanceof DecimalType) {
+                final DecimalType temp = (DecimalType) command;
+                controller.sendCommand(String.format("temp %s %s", uid, temp));
+            } else if (channel.endsWith(MODE) && command instanceof StringType) {
+                // the mode value in the command is the actual CoolMasterNet protocol command
+                controller.sendCommand(String.format("%s %s", command, uid));
+            } else if (channel.endsWith(FAN_SPEED) && command instanceof StringType) {
+                controller.sendCommand(String.format("fspeed %s %s", uid, command));
+            } else if (channel.endsWith(LOUVRE) && command instanceof StringType) {
+                controller.sendCommand(String.format("swing %s %s", uid, command));
+            } else if (command instanceof RefreshType) {
+                refresh();
             }
-        } catch (CoolMasterClientError e) {
-            logger.error("Failed to set channel {} -> {}: {}", channel, command, e.getMessage());
+        } catch (final IOException ioe) {
+            logger.error("Failed to set channel {} -> {}: {}", channel, command, ioe.getMessage());
```

Comment:
The error logging level should be reserved for issues catastrophic enough to threaten the operation of openHab itself. An error in a binding shouldn't qualify for this. Please read the [openHab coding guidelines](https://www.openhab.org/docs/developer/guidelines.html#f-logging) for expected use of logging in bindings.

Downgrade this to a warning instead.
```suggestion
            logger.warn("Failed to set channel {} -> {}: {}", channel, command, ioe.getMessage());
```